MediaBrowser.Model/System/FolderStorageInfo.cs

```diff
@@ -0,0 +1,32 @@
+namespace MediaBrowser.Model.System;
+
+/// <summary>
+/// Contains information about a specific folder.
+/// </summary>
+public class FolderStorageInfo
+{
+    /// <summary>
+    /// Gets the path of the folder in question.
+    /// </summary>
+    public required string Path { get; init; }
+
+    /// <summary>
+    /// Gets the free space of the underlying storage device of the <see cref="Path"/>.
+    /// </summary>
+    public long FreeSpace { get; init; }
+
+    /// <summary>
+    /// Gets the used space of the underlying storage device of the <see cref="Path"/>.
+    /// </summary>
+    public long UsedSpace { get; init; }
+
+    /// <summary>
+    /// Gets the kind of storage device of the <see cref="Path"/>.
+    /// </summary>
+    public string? StorageType { get; init; }
```

Comment:
Would it make sense to add an enum for this or is that too much maintenance?

Reply:
the source is already an string, so god knows what values we might get there and for now its really just a nice to know info.

Reply:
Ok so string it is then

Reply:
OK slight correction, its __supposed__ to be a fixed number of values, but those also differ from OS to OS